### ISSUE TITLE
Changeable tasklist widget

### ISSUE THREAD

**Issue by jovanmaric** [lcpz/lain#375]
Currently the default `awful.widget.tasklist` is quite limited. In the past it was possible to only display icons, cause of the changes in 3.5 it isn't anymore.The features that i'm missing from the regular tasklist are:

1. Show only icons
2. Icon themes 

Also some of the icons are either missing (termite) or rendered in low quality (atom). It would be nice to have a widget that shows the current tasks and provides styling options eg: add own icon set, choose which icons for what program

Wonder how u guys think about this.

**Comment by lcpz** [comment]
> The features that i'm missing from the regular tasklist are:

Are? Seems you forgot to add the list.

> Wonder how u guys think about this.

If you think this is a feature regression, why not working on Awesome itself? @Elv13 @psychon @actionless

**Comment by lcpz** [comment]
> (by the way: No idea what "add own icon set" means exactly)

Maybe he meant icon themes.

**Comment by alfunx** [comment]
For termite to show an icon you have to set `icon_name = terminal` in the options section of your termite configuration.

**Comment by actionless** [comment]
>  add own icon set

there is `beautiful.icon_theme` parameter, but so far i think only `menubar` honors that value

**Comment by jovanmaric** [comment]
> Are? Seems you forgot to add the list.

Whoops! edited now.

> If you think this is a feature regression, why not working on Awesome itself?

I thought it would be a nice new widget for lain, as (i think) most people use this
library to get their proper widgets for awesome, and it would complement the set
of handy working widgets in the wibar.

> Either tasklist_disable_task_name = true in your theme or disable_task_name = true in the argument table to the tasklist (there is also the same for icon).

Missed this from the docs for some reason, thanks. The only thing missing is that there
is no focus highlight, so ill either have to wait for icon themes or when something like 
font-awesome is possible in combination with font-color changes on focus.

> Do awesome.set_preferred_icon_size(42) where 42 is about the size of your tasklist. 

Have to research this, as it does not change anything regarding the icon quality. Maybe 
atom does not supply a proper bigger icon size.

> Maybe he meant icon themes.

I meant icon themes yes. For instance, being able to configure font awesome icons per 
application.

> For termite to show an icon you have to set icon_name = terminal in the options section of your termite configuration.

Have to research this as it doesn't do anything.

**Comment by lcpz** [comment]
@jovanmaric Then please do you researches and tell us what is actually missing and what is not.

**Comment by jovanmaric** [comment]
> @jovanmaric Then please do you researches and tell us what is actually missing and what is not.

I've done my research. The icon didn't show simply because I didn't have an icon for termite, namely the fallback hicolor-icon-theme didn't have one. So by installing a new icon theme (numix) it does show. 

For some reason "awesome.set_preferred_icon_size(42) " doesn't do anything, but considering that all icons look fine now i'm not spending any more time on it. I've managed to edit the tasklist with configuration in awesome. For the people interested, I have the following settings.

```
awful.screen.connect_for_each_screen(function(s)
  -- Wallpaper
  set_wallpaper(s)

  -- Each screen has its own tag table.
  awful.tag({ "", "", "" }, s, awful.layout.suit.tile)

  -- Create a promptbox for each screen
  s.mypromptbox = awful.widget.prompt()

  -- Create a taglist widget
  s.mytaglist = awful.widget.taglist(s, awful.widget.taglist.filter.all, taglist_buttons)

  -- Create a tasklist widget
  s.mytasklist = awful.widget.tasklist(s, awful.widget.tasklist.filter.currenttags, tasklist_buttons)

  -- Create the wibox
  s.mywibox = awful.wibar({ position = theme.position, screen = s, height = theme.height })

  -- Tasklist container
  local tl = wibox.container.background(s.mytasklist, theme.bg_normal, gears.shape.rectangle)
  local tasklist = wibox.container.margin(tl, 2, 2, 3, 3)
  -- local tasklist = wibox.container.place(pl, {halign="center"})

  -- Add widgets to the wibox
  s.mywibox:setup {
    layout = wibox.layout.align.horizontal,
    expand = "outside",
    {
      -- Left widgets
        layout = wibox.layout.fixed.horizontal,
        s.mytaglist,
        s.mypromptbox,
    },
      -- Middle widget
      tasklist,
    {
    -- Right widgets
      layout = wibox.layout.fixed.horizontal,
      theme.systray,
      theme.spr_left,
      theme.volume,
      theme.battery,
      theme.clock,
      theme.spr_right
    },
  }
end)
```
I still have to get the right part to align properly, as can be seen in [https://stackoverflow.com/questions/47847305/center-position-tasklist-when-using-layout-fixed](url) however i'm close to getting the preferred styling. 

What still is missing is the highlight of the current window i'm in. Ill have to wait for when this gets supported in a better way. It would be nice to setup fonts(like fontawesome) as theme-packs and simply change the font-color when that window is selected. However until then ill have to figure out a way to swap icons when the window gets selected.

In short, nothing is missing per-se, but I think that it would be nice to have a simple and configurable widget on top of awesome wm.

**Comment by lcpz** [comment]
> In short, nothing is missing per se, but I think that it would be nice to have a simple and configurable widget on top of awesome wm.

Surely we can write a [utility function](https://github.com/lcpz/lain/wiki/Utilities) which can simplify the process of dealing with Awesome APIs.

Please, give me an usage example you would like to have for such function.

**Comment by lcpz** [comment]
Closing for inactivity.

**Comment by Elv13** [comment]
Actually, it's solved. As of v4.3, it is possible to use custom widgets template in the taglist and tasklist.

**Comment by lcpz** [comment]
Even better.
